apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/PgTests.cs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.ClientTesting;
+using Google.Cloud.Spanner.Data.CommonTesting;
+using Google.Cloud.Spanner.V1;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests;
+
+[Collection(nameof(AllTypesTableFixturePostgreSql))]
+[CommonTestDiagnostics]
+[Trait(Constants.SupportedOnEmulator, Constants.No)]
+public class PgTests
+{
+    private readonly AllTypesTableFixturePostgreSql _fixture;
+    private string _lastKey;
+
+    public PgTests(AllTypesTableFixturePostgreSql fixture) =>
+        _fixture = fixture;
+
+    // Binding tests for PostgreSQL specific types. GSQL and common types are tested in the BindingTests.cs.
+
+    [Fact]
+    public async Task BindPgNumeric() =>
+        await BindingTests.TestBindNonNull(SpannerDbType.PgNumeric, PgNumeric.Parse("1.0"), GetSelectCommand, r => r.GetPgNumeric(0));
+
+    [Fact]
+    public async Task BindPgNumericArray() =>
+        await BindingTests.TestBindNonNull(SpannerDbType.ArrayOf(SpannerDbType.PgNumeric), new PgNumeric?[] { PgNumeric.Parse("0.0"), null, PgNumeric.Parse("1.0") }, GetSelectCommand);
+
+    [Fact]
+    public async Task BindPgNumericEmptyArray() =>
+        await BindingTests.TestBindNonNull(SpannerDbType.ArrayOf(SpannerDbType.PgNumeric), new PgNumeric[] { }, GetSelectCommand);
+
+    private SpannerCommand GetSelectCommand(SpannerDbType parameterType, object value)
```

Comment:
Maybe rename this one `GetSelectCommandForBinding`.
Although once these return the reader, it might be clearer, something like GetBindingTestReader or similar.